### ISSUE TITLE
`Type` and `Val` are missing `SourceIndex`

### ISSUE THREAD

**Issue by tdroxler** [alephium/ralph-lsp#156]
As mentioned in [this comment](https://github.com/alephium/ralph-lsp/issues/152#issuecomment-2006886667)

**Comment by simerplaha** [comment]
Just noting that `SourceIndex` on `Type` is needed to provide jump definitions for [primitive types](https://docs.alephium.org/ralph/getting-started/#primitive-types).

**Comment by tdroxler** [comment]
yes but primitives types are case object and we can't apply them the `SourceIndex` yet

**Comment by simerplaha** [comment]
This will not be needed after the switch to `SoftAST`  is complete.
